python/ray/serve/controller.py

```diff
@@ -1114,20 +981,16 @@ async def update_backend_config(self, backend_tag: BackendTag,
             # Scale the replicas with the new configuration.
 
             # This should be to run the control loop
-            self.actor_reconciler._scale_backend_replicas(
-                self.backend_state.backends, backend_tag,
-                backend_config.num_replicas)
+            self.backend_state.scale_backend_replicas(
+                backend_tag, backend_config.num_replicas)
 
             # NOTE(edoakes): we must write a checkpoint before pushing the
             # update to avoid inconsistent state if we crash after pushing the
             # update.
             self._checkpoint()
 
-            # Inform the routers about change in configuration
-            # (particularly for setting max_batch_size).
-
-            await self.actor_reconciler._enqueue_pending_scale_changes_loop(
-                self.backend_state)
+            # Inform the routers and backend replicas about config changes.
+            self.notify_backend_configs_changed()
```

Comment:
Oh, good call re-adding this. Can you expand the comment to state that this is necessary because an updated backend config _may not enqueue a change in the number of replicas_

Reply:
This is actually the only place we need to notify this change, I removed it from the control loop. New replicas that are created will fetch the config themselves, no need to notify in that case.